### PR TITLE
Edit Python: File-Methods/Add Term Entry Python: Built-in Functions: `open()`

### PR DESCRIPTION
**Pull request by shiningflutes**
### Description

Added a new entry on files in python as per issue #273 

### Type of Change
- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by Dusch4593** [comment]
Hey @shiningflutes! I'll take a look at your entry for the first review and tag you when I'm finished. It should be either today or tomorrow. 😄

File: content/python/concepts/file-methods/files.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'Files'
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/code-foundations'
+  - 'paths/computer-science'
+  - 'paths/front-end-engineer-career-path'
```
Comment (Dusch4593) [inline content/python/concepts/file-methods/files.md:15]:
```suggestion
  - 'learn-python-3'
  - 'paths/computer-science'
```

Just two links for `CatalogContent` will do.

**Comment by Dusch4593** [comment]
Hey @shiningflutes! I just got around to having a look at this PR, and I discovered a few things:

- There seems to be [a Python entry about files and file methods](https://www.codecademy.com/resources/docs/python/file-methods) already published on the Docs site. Perhaps #273 is older and slipped past us when the entry was published.
- [The file associated with that published entry](https://github.com/Codecademy/docs/blob/main/content/python/concepts/file-methods/file-methods.md) actually lives in the `python/concepts/file-methods/` directory.

Perhaps what we can do is take some of what you contributed to be incorporated from your `file.md` file to the [`file-methods.md` file](https://github.com/Codecademy/docs/blob/main/content/python/concepts/file-methods/file-methods.md). Here are few suggestions for doing that:

- Add your code snippet example (the one from the "Methods" section) near the end of the `file-methods.md`.
- Incorporate the section about working with `t` (text) and `b` (binary) file formats in Python.

After you've made those changes in `file-methods.md`, **make sure to delete `files.md` before pushing the new changes**. 

Sorry again about this. I'm happy to help where I can and answer any questions 😄

**Comment by KTom101** [comment]
@Dusch4593 I also mentioned this in the issue but perhaps the first half of this entry can be used to satisfy [issue #527 ](https://github.com/Codecademy/docs/issues/527).

**Comment by Dusch4593** [comment]
> @Dusch4593 I also mentioned this in the issue but perhaps the first half of this entry can be used to satisfy [issue #527 ](https://github.com/Codecademy/docs/issues/527).

Absolutely! Since there is already a "File Methods" Python entry in Docs most, if not all, of what's in this PR can be incorporated both in that entry and in, as you pointed out, material from the first half can be used for a separate `.open()` term entry. 😃

**Comment by Dusch4593** [comment]
Hi @shiningflutes!

Given that some time has passed, I made the following changes in 69b17d1ebc43384cb7d615f9d543109b19347539, 62deda756aebd6cbe977f3490632ead275d3a64f, and fbf76083f3b7d9067d331215e6dbc5c58e31897f to set us up for continuing work on this PR and merging your contributions to Docs 😄 :

- Created a new term entry, `open.md`, for the [Python: Built-in Functions concept entry](https://www.codecademy.com/resources/docs/python/built-in-functions) and moved all of the relevant content you wrote in `files.md` to that new file.
- Took the rest of what I could to your `files.md` file and incorporated into the existing `file-methods.md` file.
  - This includes your code snippet which ended up needing a rewrite because I couldn't get the expected output as it was written. What I like to do sometimes is check that the code in any files, snippets, or examples are functional and returning the expected output.
 - I deleted the original `files.md` file and renaming the existing `file-methods.md` file but was running into some checks failing on GitHub's end after pushing. For now, we'll keep working with `file-methods/file-methods.md` but I think this should ultimately be changed to `files/files.md` in a separate PR by someone on the Codecademy Free Team.

Going forward, I think that the `file-methods.md` file is solid enough that it needs few to no further edits. But what we can do, @shiningflutes, is have you finish making edits in the new `open.md` file in the `concepts/python/built-in-functions/terms/` directory, including:

- Add a value for the `Description` meta-property near the top (see [this term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md?plain=1) for guidance on writing it).
- Mention how the `r` read-only mode is the default mode when using `open()`.
- Write a "## Syntax" section with 1-2 sentences and a code snippet that highlights the two syntaxes for `open()`:

```
first_syntax = open("example.txt")
first_syntax.close()

with open("example.txt")
  print("This syntax auto-closes the file.")
```

- Create an "## Example" section that further highlights the `open()` method.

cc: @KTom101

**Comment by Dusch4593** [comment]
@shiningflutes Everything going ok with the `.open()` entry?

**Comment by shiningflutes** [comment]
@Dusch4593 no issues currently :) Sorry for the late reply as I am rather busy currently. Could I ask how do I check the expected output? Since we are dealing with a file that is not actually created in the example. Not sure what the expected output should be. Also, would you like a different example for both ##syntax and ##example (your last 2 points), as they seem overlapping to me if I've already illustrated an example in ##syntax.

**Comment by Dusch4593** [comment]
Hey @shiningflutes! No need to be sorry :)

> Could I ask how do I check the expected output? Since we are dealing with a file that is not actually created in the example. Not sure what the expected output should be.

If we're talking about the "Syntax" section for the `open()` entry, there's no need to expect a value; we should use a \```pseudo code snippet for that:

```pseudo
# First syntax
f = open("file-name.string")
f.close()

# Second syntax
with open("file-name.string")
  print("This syntax auto-closes the file.")
```

If we're talking about the "Example" section that I recommended for the `open()` entry, a few notes:

* The example from your changes in the `file-methods.md` entry (lines 25-54) should be valid:
  * Creates, opens, writes to, and closes a file.
  * Reopens, reads, prints content from, and closes that same file.

* I suggest moving that "Example" section (lines 25-54) from `file-methods.md` to `open.md` (below the "Opening files" section"). 

Hopefully this answers your questions and feel free to ask more for clarity. Always happy to help 😄 I'd say after you move that "Example" section and create the "Syntax" section in `open.md`, I'll be ready to take a look for a formal review.

**Comment by yangc95** [comment]
This PR looks like it's in need of some TLC! @shiningflutes really appreciate the work put into these with everything else that may be going on.

@Dusch4593 hope you don't mind me taking a look at this as-is and helping out with some improvements(comments/suggestions) using your provided feedback. ☺️

**Review by yangc95 — COMMENTED** [review]
Included are some things to work on for these two entries. Hope it's not too much, but also let us know if you need further assistance.

Please let us know as well if you're alright with us (maintainers) making direct changes to your branch to get this incorporated into Docs 🙏

File: content/python/concepts/built-in-functions/terms/open/open.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+There are several modes Python can do when opening a file. The default one is reading. Here is a list of modes.
+
+r - reads a file\
+w - writes a file\
+a - appends a file\
+x - creates a file
+
+Files can be read in either text or binary, and are denoted as `t` and `b` respectively. The default is text.
+
+```py
```
[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/python/concepts/built-in-functions/terms/open/open.md:28]: That's a _great_ idea! 🚀

File: content/python/concepts/built-in-functions/terms/open/open.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  - 'Files'
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+## Opening files
+
+In Python, the `open()` function is built-in, and can be used to open files. To open a file, the file name can be specified in the function. Do note that the entire file path has to be specified unless the file is in the current directory.
```
[the remark this answers is not in the document]
Comment (yangc95) [reply content/python/concepts/built-in-functions/terms/open/open.md:17]: Disregard this

**Comment by yangc95** [comment]
Hi @shiningflutes 👋 hope all is well! Wondering how everything is going and seeing if you needed some direct assistance with this PR. LMK!

**Comment by Dusch4593** [comment]
@shiningflutes @yangc95 

Its been a while since we've seen action on this one and wanted to help out with some of the following changes in 733864044c1b1e725e371b422b438d28102bd733:

* In `open.md`:
  * Removed "Opening Files" section header and added "Syntax".
  * Added a meta-`Description` up top. 
  * Added a Codebyte Example with a runnable code snippet (_highly_ recommend if [possible](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md#codebytes-runnable-code-blocks) and if you're up for a challenge 😄).
  * Restructured and reworded the list of common file modes.
 
* In `files.md`:
  * Super light edits.

**Review by Dusch4593 — APPROVED** [review]
@shiningflutes Approving for my part but let's finish the remaining comments/suggestions left by @yangc95

**Review by yangc95 — APPROVED** [review]
@shiningflutes I've also made some direct changes due to the time that has passed on this entry. @Dusch4593 I think this is looking good for merge 👍

**Comment by yangc95** [comment]
Live links:
 
https://www.codecademy.com/resources/docs/python/files
https://www.codecademy.com/resources/docs/python/built-in-functions/open

TY @shiningflutes @Dusch4593 @KTom101 

P.S. If you wanted to appear a a contirbutor for this entry, make sure to link your GitHub to your Codecademy account. Otherwise, everything should be good to go ~

**Comment by shiningflutes** [comment]
@yangc95 @Dusch4593 
I'm so sorry for the delay! Was really busy with school and this week's my holiday.
Is there anything else I can help with?

Thank you for the help

**Comment by yangc95** [comment]
> @yangc95 @Dusch4593 I'm so sorry for the delay! Was really busy with school and this week's my holiday. Is there anything else I can help with?
> 
> Thank you for the help thus far, I would be glad to help in the future if help is needed (perhaps when my school term ends..) :)

Hey @shiningflutes, no biggie~ Hope you've been well ☺️ studies first for sure!

When you have some time, I'd love to hear your story on contributing to Docs [Forum - What made you want to contribute to Docs](https://discuss.codecademy.com/t/what-made-you-want-to-contribute-to-docs/663989/4)!

**Pull request merged**